secant/Features/BackupFlow/Views/RecoveryPhraseBackupValidationView.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  RecoveryPhraseBackupView.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct RecoveryPhraseBackupValidationView: View {
+    let store: RecoveryPhraseValidationStore
+
+    var body: some View {
+        WithViewStore(self.store) { viewStore in
+            VStack {
+                VStack {
+                    Text("Drag the words below to match your backed-up copy.")
+                        .bodyText()
+
+                    viewStore.state.missingWordGrid()
+                        .padding(.horizontal, 30)
+                }
+                .padding(.horizontal)
+                .padding(.bottom, 10)
+
+                VStack(spacing: 20) {
+                    let state = viewStore.state
+                    let chunks = state.phrase.toChunks()
+                    ForEach(Array(zip(chunks.indices, chunks)), id: \.0) { index, chunk in
+                        WordChipGrid(chips: state.wordChips(for: index, groupSize: RecoveryPhraseValidationState.wordGroupSize, from: chunk))
+                            .background(Asset.Colors.BackgroundColors.phraseGridDarkGray.color)
+                            .whenIsDroppable(!state.groupCompleted(index: index), dropDelegate: state.dropDelegate(for: viewStore, group: index))
+                    }
+                }
+                .padding()
+                .background(Asset.Colors.BackgroundColors.phraseGridDarkGray.color)
+
+            }
+            .applyScreenBackground()
+        }
+        .navigationBarTitleDisplayMode(.inline)
+        .navigationTitle(Text("Verify Your Backup"))
+    }
+}
+
+private extension RecoveryPhraseValidationState {
+    @ViewBuilder func missingWordGrid() -> some View {
+        let columns = Array(repeating: GridItem(.flexible(minimum: 40, maximum: 120), spacing: 20), count: 2)
+        LazyVGrid(columns: columns, alignment: .center, spacing: 20 ) {
+            ForEach(0..<missingWordChips.count) { chipIndex in
+                PhraseChip(kind: missingWordChips[chipIndex])
+                    .makeDraggable()
+                    .frame(
+                        minWidth: 0,
+                        maxWidth: .infinity,
+                        minHeight: 30
+                    )
+            }
+        }
+        .padding(0)
+    }
+}
+
+extension RecoveryPhraseValidationState{
```

Comment:
These functions seems very similar, and duplicate quite a bit of logic.  Is it possible to condense this down to just a single function instead of three?